sentry-android-core/src/main/java/io/sentry/android/core/adapters/OrientationDeserializerAdapter.java

```diff
@@ -0,0 +1,20 @@
+package io.sentry.android.core.adapters;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import io.sentry.core.protocol.Device;
+import java.lang.reflect.Type;
+import java.util.Locale;
+
+public class OrientationDeserializerAdapter implements JsonDeserializer<Device.DeviceOrientation> {
+  @Override
+  public Device.DeviceOrientation deserialize(
+      JsonElement json, Type typeOfT, JsonDeserializationContext context)
+      throws JsonParseException {
+    return json == null
+        ? null
+        : Device.DeviceOrientation.valueOf(json.getAsString().toUpperCase(Locale.ROOT));
```

Comment:
if the value in the JSON doesn't match `DeviceOrientation` is this going to throw?
Should we be supressing exception in adapters like this?

Reply:
guess so, for all of them

Reply:
Yeah I think that'll help debugging the library a lot when weird stuff happens and we get support tickets.

Reply:
created an internal task for that